### ISSUE TITLE
Improve HitRate Plot

### ISSUE THREAD

**Issue by LilySnow** [DeepRank/deeprank#4]
**My experiment setting**: 3D conv, dict_filter = {'DOCKQ':'>0.1'}, feature with pssm_ic only, regression
 
And I got the following strange hit rate plot:
![image](https://user-images.githubusercontent.com/8004705/36890651-f9a8e454-1dfe-11e8-88b9-9e2d1bf228fa.png)

Here is the scatter plot of predicted DockQ vs. real DockQ:
![image](https://user-images.githubusercontent.com/8004705/36890680-1ea0685e-1dff-11e8-8511-78aee70638e5.png)

**My code (also can be found on alembick: /home/lixue/deeprank/BM4/xper/all/exp001/learn.py and model3d.py):** 

> from deeprank.learn import *
> import model3d
> import torch.optim as optim
> import os
> import numpy as np
> import glob
> # declare the dataset instance
> 
> path = '/home/deep/projects/deeprank/BM4/'
> 
> test_files = ['1E6E.hdf5']
> train_database = [ f for f in glob.glob(path + '*.hdf5') if '1E6E.hdf5' not in f]
> test_database = [path+f for f in test_files]
> 
> data_set = DataSet(train_database,
>                 test_database = test_database,
>                 grid_shape = (30,30,30),
>                   # select_feature= 'all',
>                 select_feature = {'Feature_ind' : ['pssm_ic']},
>                 pair_chain_feature=np.add,
>                 select_target='DOCKQ',
>                 # does dict_filter filter only  the training data
>                 dict_filter = {'DOCKQ':'>0.1'},
>                 normalize_features=True,normalize_targets=True,tqdm=False)
> 
> 
> # create the network
> model = NeuralNet(data_set, model3d.cnn,cuda=True,ngpu=1,plot=True)
> 
> # change the optimizer (optional)
> model.optimizer = optim.SGD(model.net.parameters(),
>         lr=0.001,momentum=0.9,weight_decay=0.005)
> 
> # start the training
> model.train(nepoch = 30,train_batch_size = 50,num_workers=8,divide_trainset=[0.8,0.2])
> model.save_model()

**The NN model file:**

> import torch
> from torch.autograd import Variable
> import torch.nn as nn
> import torch.nn.functional as F
> import torch.optim as optim
> import torch.utils.data as data_utils
> 
> 
> 
> '''
> definition of the Convolutional Networks
> 
> The model must take as an argument the input shape
> This allows to automaically precompute
> the input size of the first FC hidden layer
> 
> '''
> 
> class cnn(nn.Module):
> 
> 	def __init__(self,input_shape):
> 		super(cnn,self).__init__()
> 
> 		self.conv1 = nn.Conv3d(input_shape[0],4,kernel_size=2)
> 		self.pool  = nn.MaxPool3d((2,2,2))
> 		self.conv2 = nn.Conv3d(4,5,kernel_size=2)
> 		self.conv2_drop = nn.Dropout3d()
> 
> 		size = self._get_conv_output(input_shape)
> 
> 		self.fc1   = nn.Linear(size,84)
> 		self.fc2   = nn.Linear(84,1)
> 
> 		self.sm = nn.Softmax()
> 
> 	def _get_conv_output(self,shape):
> 		inp = Variable(torch.rand(1,*shape))
> 		out = self._forward_features(inp)
> 		return out.data.view(1,-1).size(1)
> 
> 	def _forward_features(self,x):
> 		x = self.pool(F.relu(self.conv1(x)))
> 		x = self.pool(F.relu(self.conv2(x)))
> 		return x
> 
> 	def forward(self,x):
> 
> 		x = self._forward_features(x)
> 		x = x.view(x.size(0),-1)
> 		x = F.relu(self.fc1(x))
> 		x = self.fc2(x)
> 		return x

**Comment by LilySnow** [comment]
The running time is 8 min.

**Comment by LilySnow** [comment]
The seemingly strange hit rate plot is caused by the extremely long x-axis. We should zoom in to top 400.

**Comment by NicoRenaud** [comment]
According to Li : The x-axis of the hit rate plot should be 0 to the number of test models

**Comment by CunliangGeng** [comment]
@LilySnow Has this issue been solved since it was posted last March?

**Comment by LilySnow** [comment]
It is solved. We are now using percentage as our x-axis.
